### ISSUE TITLE
http client&& coroutine http client 的$http->set(['keep_alive' => false]);是否无效？？

### ISSUE THREAD

**Issue by 8923052** [swoole/swoole-src#1670]
Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

```
$http = new swoole_http_server("127.0.0.1", 9501);
$http->on('workerStart', function (swoole_server $server, $worker_id) {
	if($worker_id == 0) {
		$cli = new \Swoole\Coroutine\Http\Client('127.0.0.1', 9501);
		
		$header = [ 
				//'Connection' => 'close', // 未复用连接时，关闭长连接
				'Keep-alive'=>60,
				"User-Agent" => 'Chrome/49.0.2587.3',
				'Accept' => 'text/html,application/xhtml+xml,application/xml,application/json',
				'Accept-Encoding' => 'gzip' 
		];
		//$cli->setHeaders($header);
		$cli->set([
				'keep_alive' => false,
				'timeout' => 5
		]);
		$cli->setMethod('GET');
		$cli->setData([ ]);
		$cli->execute('/');
		var_dump('$cli->setting', $cli->setting);
		var_dump('$cli->headers', $cli->headers);
		var_dump('$cli->body', $cli->body);
	}
});
$http->on('request', function ($request, $response) {
	var_dump('$request->header', $request->header);
	$response->end("<h1>Hello Swoole</h1>");
});
$http->start();
```

2. What did you expect to see?
```
string(16) "$request->header"
array(5) {
  ["connection"]=>
  string(10) "keep-alive"
  ["host"]=>
  string(9) "127.0.0.1"
  ["accept-encoding"]=>
  string(4) "gzip"
  ["content-type"]=>
  string(33) "application/x-www-form-urlencoded"
  ["content-length"]=>
  string(1) "0"
}
string(13) "$cli->setting"
array(2) {
  ["keep_alive"]=>
  bool(false)
  ["timeout"]=>
  float(5)
}
string(13) "$cli->headers"
array(5) {
  ["server"]=>
  string(18) "swoole-http-server"
  ["content-type"]=>
  string(9) "text/html"
  ["connection"]=>
  string(10) "keep-alive"
  ["date"]=>
  string(29) "Wed, 30 May 2018 08:44:16 GMT"
  ["content-length"]=>
  string(2) "21"
}
string(10) "$cli->body"
string(21) "<h1>Hello Swoole</h1>"
```

3. What did you see instead?

按理说设置了keep_alive为false, request的header中，Connection应该为close，实际仍然是keep-alive

设置'Keep-alive'=>60,再set(['keep_alive' => false])，request header也没有变化。

4. What version of Swoole are you using (`php --ri swoole`)?

php --ri swoole

swoole

swoole support => enabled
Version => 4.0.0-beta
Author => tianfeng.han[email: mikan.tenny@gmail.com]
coroutine => enabled
epoll => enabled
eventfd => enabled
timerfd => enabled
signalfd => enabled
spinlock => enabled
rwlock => enabled
async redis client => enabled
async http/websocket client => enabled
sockets => enabled
openssl => enabled
Linux Native AIO => enabled
pcre => enabled
zlib => enabled
mutex_timedlock => enabled
pthread_barrier => enabled
futex => enabled
mysqlnd => enabled
jemalloc => enabled

Directive => Local Value => Master Value
swoole.aio_thread_num => 2 => 2
swoole.display_errors => On => On
swoole.use_namespace => On => On
swoole.use_shortname => On => On
swoole.fast_serialize => Off => Off
swoole.unixsock_buffer_size => 8388608 => 8388608

5. What is your machine environment used (including version of kernel & php & gcc) ? 
php -v
PHP 7.2.6 (cli) (built: May 26 2018 08:53:02) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies

uname -a
Linux localhost.localdomain 3.10.0-693.5.2.el7.x86_64 #1 SMP Fri Oct 20 20:32:50 UTC 2017 x86_64 Linux

gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-alpine-linux-musl/6.4.0/lto-wrapper
Target: x86_64-alpine-linux-musl
Configured with: /home/buildozer/aports/main/gcc/src/gcc-6.4.0/configure --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --build=x86_64-alpine-linux-musl --host=x86_64-alpine-linux-musl --target=x86_64-alpine-linux-musl --with-pkgversion='Alpine 6.4.0' --enable-checking=release --disable-fixed-point --disable-libstdcxx-pch --disable-multilib --disable-nls --disable-werror --disable-symvers --enable-__cxa_atexit --enable-default-pie --enable-cloog-backend --enable-languages=c,c++,objc,java,fortran,ada --disable-libssp --disable-libmpx --disable-libmudflap --disable-libsanitizer --enable-shared --enable-threads --enable-tls --with-system-zlib --with-linker-hash-style=gnu
Thread model: posix
gcc version 6.4.0 (Alpine 6.4.0)
6. If you are using ssl, what is your openssl version?

openssl version
OpenSSL 1.0.2o  27 Mar 2018

**Comment by matyhtf** [comment]
设置为 keep_alive = 0 或 keep_alive = 1 试试。

**Comment by twose** [comment]
的确有问题, 已修复, 多谢反馈
fixed in #1673
